src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -1064,7 +1071,7 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 			quickPick.value = value;
 		}
 		return new Promise<void>(r => {
-			quickPick.show();
+			this._instantiationService.invokeFunction(formatPinnedItems, runRecentStorageKey, quickPick, async () => await this.runRecent(type, filterMode, value));
 			this._terminalInRunCommandPicker.set(true);
 			quickPick.onDidHide(() => {
 				this._terminalInRunCommandPicker.set(false);
```

Comment:
The callback here will probably mess with the `_terminalInRunCommandPicker` context key.

Reply:
not sure what you mean, but moved it above that so think it should be fine

Reply:
I think I was wrong actually, I thought this was outside `runRecent`. Should be fine but you can verify by pinning something and then using the ctrl+c keybinding here:

https://code.visualstudio.com/docs/terminal/shell-integration#_run-recent-command